src/components/settings/SettingsLMSTab/LMSConfigs/utils.tsx

```diff
@@ -0,0 +1,185 @@
+
+type ConfigCamelCase = {id?: string, uuid?: string, clientId?: string} |
+  BlackboardConfigCamelCase | CanvasConfigCamelCase | CornerstoneConfigCamelCase | DegreedConfigCamelCase | MoodleConfigCamelCase | SAPConfigCamelCase;
+type ConfigSnakeCase = {enterprise_customer?: string, active?: boolean} |
+  BlackboardConfigSnakeCase | CanvasConfigSnakeCase | CornerstoneConfigSnakeCase | DegreedConfigSnakeCase | MoodleConfigSnakeCase | SAPConfigSnakeCase;
+
+export const LMS_AUTHORIZATION_FAILED = "LMS AUTHORIZATION FAILED";
+
+export async function handleSubmitHelper(
+  enterpriseCustomerUuid: string,
+  transformedConfig: ConfigSnakeCase,
+  existingData: any,
+  onSubmit: (param: ConfigCamelCase) => void,
+  formFieldsChanged: Boolean,
+  currentFormFields: any,
+  lmsType: string,
+  channelMap: Record<string, Record<string, any>>,
+  errHandler: FormWorkflowErrorHandler | undefined,
+  dispatch: any,
+) {
+  transformedConfig.enterprise_customer = enterpriseCustomerUuid;
+  let err = "";
+  if (formFieldsChanged) {
+    if (currentFormFields?.id) {
+      try {
+        transformedConfig.active = existingData.active;
+        const response = await channelMap[lmsType].update(transformedConfig, existingData.id)
+        currentFormFields = camelCaseDict(
+          response.data
+        ) as ConfigCamelCase;
+        onSubmit(currentFormFields);
+        dispatch?.(updateFormFieldsAction({ formFields: currentFormFields }));
+      } catch (error) {
+        err = handleErrors(error);
+      }
+    } else {
+      try {
+        transformedConfig.active = false;
+        const response = await channelMap[lmsType].post(transformedConfig)
+        currentFormFields = camelCaseDict(
+          response.data
+        ) as ConfigCamelCase;
+        onSubmit(currentFormFields);
+        dispatch?.(updateFormFieldsAction({ formFields: currentFormFields }));
+      } catch (error) {
+        err = handleErrors(error);
+      }
+    }
+  }
+  const authorizeError = await handleSubmitAuthorize(lmsType, existingData, currentFormFields, channelMap, dispatch);
+  if (err) { errHandler?.(err); }
+  if (authorizeError) { errHandler?.(authorizeError); }
+
+  return currentFormFields;
+}
+
+async function handleSubmitAuthorize(
+  lmsType: string, 
+  existingData: ConfigCamelCase,
+  currentFormFields: any,
+  channelMap: Record<string, Record<string, any>>,
+  dispatch: any,
+) {
+  if ((lmsType === BLACKBOARD_TYPE || lmsType === CANVAS_TYPE) && currentFormFields && !currentFormFields?.refreshToken) {
+    let oauthUrl: string;
+    if (lmsType == BLACKBOARD_TYPE) {
+      let appKey = existingData.clientId;
+      let configUuid = existingData.uuid;
+      if (!appKey || !configUuid) {
+        try {
+          if (lmsType === BLACKBOARD_TYPE) {
+            const response = await channelMap[lmsType].fetchGlobal();
+            appKey = response.data.results[response.data.results.length - 1].app_key;
```

Comment:
nit: Can use [].at() with negative indexing for a less verbose statement
```suggestion
            appKey = response.data.results.at(-1).app_key;
```